examples/pos/withstatement.effekt

```diff
@@ -11,8 +11,8 @@ def bar { f: (Int, String) => Unit / {}}: Unit =
 
 def user(): Unit = {
     with printer;
-    with val x: Int = foreach([1,2,3]);
-    with val (a, b) = bar;
+    with val x = foreach([1,2,3]);
+    with val a, b = bar;
```

Comment:
This is a breaking change ^ as type annotations are no longer supported and multiple arguments have to be written _without_ the parenthesis.

Reply:
Does it make sense to support `else` clauses for exhaustive pattern matching?

Like:

```
with val Some(v) = foo() else do raise()
...
```

Reply:
I don't see a reason why not :)
In an ideal case, I'd like to share as much parser code as possible with destructuring ala `val (x, Some(y)) = ...` (and with lambda-case)